spec/acceptance/enum_traits_spec.rb

```diff
@@ -0,0 +1,161 @@
+describe "enum traits" do
+  context "when automatically_define_enum_traits is true" do
+    it "builds traits automatically for model enum field" do
+      define_model("Task", status: :integer) do
+        enum status: { queued: 0, started: 1, finished: 2 }
+      end
+
+      FactoryBot.define do
+        factory :task
+      end
+
+      Task.statuses.each_key do |trait_name|
+        task = FactoryBot.build(:task, trait_name)
+
+        expect(task.status).to eq(trait_name)
+      end
+
+      Task.reset_column_information
+    end
+
+    it "prefers user defined traits over automatically built traits" do
+      define_model("Task", status: :integer) do
+        enum status: { queued: 0, started: 1, finished: 2 }
+      end
+
+      FactoryBot.define do
+        factory :task do
+          trait :queued do
+            status { :finished }
+          end
+
+          trait :started do
+            status { :finished }
+          end
+
+          trait :finished do
+            status { :finished }
+          end
+        end
+      end
+
+      Task.statuses.each_key do |trait_name|
+        task = FactoryBot.build(:task, trait_name)
+
+        expect(task.status).to eq("finished")
+      end
+
+      Task.reset_column_information
+    end
```

Comment:
to some extent the examples could be refactored to use a shared example, but i am not sure at what point it is just "clever" overengineering vs making it more readable. generally all the tests are doing

- define the "same" model with integer or string column, optionally adding an (AR) enum
- define the factory with or without explicit traits, obviously this is the main part, not strictly the subject of the tests but close (so the less code share available)
- assert the factory's traits building with the right attribute value

i would argue the first and last steps are very easy to share between the examples (but again, at some point it will be counter-productive), obviously low prio & more debatable

Reply:
There might be some room to pull out some shared methods here, but lately I have been preferring repetition in my tests over clever shared behavior. I think https://thoughtbot.com/blog/the-case-for-wet-tests covers a lot of what I am thinking. I also talk a little bit about my preference for duplication in tests in https://www.bikeshed.fm/186.